api/envoy/extensions/certificate_selectors/on_demand_secret/v3/config.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package envoy.extensions.certificate_selectors.on_demand_secret.v3;
+
+import "envoy/config/core/v3/extension.proto";
+import "envoy/config/core/v3/config_source.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.certificate_selectors.on_demand_secret.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/certificate_selectors/on_demand_secret/v3;on_demand_secretv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: On-demand secret certificate selector]
+// [#extension: envoy.certificate_selectors.on_demand_secret]
+
+// Fetches the secret on-demand while allowing the parent cluster or listener to accept connections
+// without warming. The certificate secret is validated against the outer common TLS context, e.g.
+// by setting FIPS compliance policy on the loaded secret material.
+message Config {
+  // Defines the source of the secrets.
+  config.core.v3.ConfigSource config_source = 1 [(validate.rules).message = {required: true}];
+
+  // Extension to specify a mapping to the secret name.
+  // [#extension-category: envoy.tls.certificate_mappers]
+  config.core.v3.TypedExtensionConfig secret_name = 2 [(validate.rules).message = {required: true}];
+
+  // A list of secret resource names to start fetching on configuration load.
+  repeated string prefetch_names = 3;
+}
+
+// A mapping to a fixed certificate name.
+message StaticName {
```

Comment:
I don't think this needs to be under the on_demand_secret package, but should be moved to some other package.